Idle_Game/app/src/main/java/com/example/idle_game/data/repositories/GameRepository.kt

```diff
@@ -46,13 +45,21 @@ class GameRepository(
                     accessToken = null
                 )
                 gameDao.insertPlayer(playerData)
+                createNewInventory()
             }
         } catch (e: HttpException) {
             onFailure()
         }
     }
 
     suspend fun signIn(username: String, password: String, onFailure: () -> Unit = {}) {
+        try {
+            if (gameDao.getPlayer().first().username != username) {
+                gameDao.updateUsername(username)
+                createNewInventory()
+            }
```

Comment:
Try Catch Block tut überhaupt gar nichts im Fehlerfall

Reply:
Das ist Richtig. Im Fehlerfall soll er einfach weiter gehen

Reply:
Also ist der Fehler erwartet?
Ansonsten kannst du den ganzen Block auch entfernen

Reply:
Es kommt da zu einem Fehler, wenn die db nicht existiert. Jedoch soll in jedem Fall die Funktion weiter ausgeführt werden. d.h: die Exception soll abgefangen werden.
Der Catchblock sagt also einfach nur: Hier gehts noch weiter.

Reply:
Dann kommt es danach doch wieder zum Fehler? 
Was ändert dieser Block also?

Reply:
Bitte entferne das einfach, das Reposititory wird schon nicht ohne room-db erstellt 😂😂

Reply:
Nein nachher wird ein player inserted.

Reply:
Ich finde es aber sinvoll :)

Reply:
Stell dir vor. Du hast einen acc und bekommst ein neues Handy (vllt sogar ein Samsung fold 👀)
Dann hast du keine db ;)

Reply:
Doch dann habe ich eine DB, die wird durch die DI erstellt.
Sonst könnte kein DAO und keine reposititory erstellt werden können

Reply:
Und was ändert es daran, dass es zum fehler kommt?

Reply:
Es kommt nicht zum Fehler kommt, ansonsten müsstest du jede Datenbank Anfrage in einen try Catch Block packen

Reply:
![image](https://github.com/user-attachments/assets/c0f486d5-56d4-4e0c-91cf-5c5ff7710da6)
Hab mal ein Log in den catch reingepackt.

Reply:
Okay jetzt verstehe ich was du meinst 👍
Der Fehler wird nicht durch das Fehlen einer DB ausgelöst, sondern dadurch dass du eine leere Tabelle durchsuchst.

Reply:
In diesem Fall ist es analog zu dem, was ich in Zeile 77 angemerkt habe